linux/LinuxProcessList.c

```diff
@@ -1542,6 +1543,15 @@ static bool LinuxProcessList_recurseProcTree(LinuxProcessList* this, openat_arg_
 
       LinuxProcessList_recurseProcTree(this, procFd, "task", proc, period);
 
+      if ((ss->flags & PROCESS_FLAG_LINUX_CGROUP) || hideRunningInContainer) {
+         LinuxProcessList_readCGroupFile(lp, procFd);
+         if (hideRunningInContainer && lp->cgroup && 
+            !(String_startsWith(lp->cgroup, "/user") || String_startsWith(lp->cgroup, "/machine") || String_startsWith(lp->cgroup, "/system")) ) {
+            if (!LinuxProcessList_checkPidNamespace((Process *)lp, procFd))
+               goto errorReadingProcess;
+         }
+      }
+
```

Comment:
Why did you move this block of code?

Reply:
Before moving the code, there was a significant lag between the moment one enables the "hide running in a container" option and the moment the change is actually seen on the screen.

Reply:
My theory is that it's because, initially, none of the processes are marked as "running in container". If the option is toggled just after this condition.

```
if ((ss->flags & PROCESS_FLAG_LINUX_CGROUP) || hideRunningInContainer)
``` 
The change will still not take effect in the next iteration of the loop, since the process will still not be marked as "running in a container" in this condition.

```
if (preExisting && hideRunningInContainer && Process_isRunningInContainer(proc))
```
It finally takes effect after one more iteration.